src/strategy/mod.rs

```diff
@@ -96,7 +94,7 @@ impl UpdateStrategy {
     }
 
     /// Check if this agent is allowed to check for updates at this time.
-    pub(crate) fn can_check_and_fetch(&self, _identity: &Identity) -> impl Future<Output = bool> {
+    pub(crate) fn can_check_and_fetch(&self) -> impl Future<Output = bool> {
```

Comment:
I think this function can be removed at all. If I'm not wrong (please double-check that), all the underlying methods always return `Ok(true)` so there is no useful logic performed here.

Reply:
The only two places that used this function are https://github.com/coreos/zincati/blob/master/src/update_agent/actor.rs#L188 and https://github.com/coreos/zincati/blob/master/src/update_agent/actor.rs#L219.

I've removed the function `can_check_and_fetch` and initialized the `ActorFuture` with a constant `actix::fut::ok::<bool, bool, Self>(true)`

Reply:
Also for both `can_finalize` and `report_steady`, strategies `Immediate` and `Periodic` had a hard-coded constant `true` as results. I didn't remove those parts for cleanness of the match block, since unlike `can_check_and_fetch`, they actually have functional values in `FleetLock`.